dual_net.py

```diff
@@ -134,6 +139,45 @@ def should_log(i):
                     logger.add_summary(weight_summaries, global_step)
             self.save_weights()
 
+    def validate(self, tf_records, batch_size=128, init_from=None, num_steps=1000):
+        """Compute only the cost components for a set of tf_records, ideally a
```

Comment:
cost components? tf_records? 

Trying to nail down jargon since our goal is to be a readable implementation.

Reply:
The cost components of our graph; tf_records are a set of tf_examples.  these are kind of standard for tensorflow, IIUC, so i'm not sure how much we need to explicitly describe them.  

Happy to elaborate a bit more here but this brings up a kind of a question about DRY-ness of the comments:  we have dual_net.DualNetworkTrainer.train, rl_loop.train, and main.train.  They're all gonna be real similar, and duplicating the docstrings seems pointless, esp. when one is just a shell around the others.

Reply:
Dual net should probably be the SOT for comments then.